src/splat.ts

```diff
@@ -84,6 +78,31 @@ class Splat extends Element {
         return true;
     }
 
+    // recalculate the local space splat aabb and update engine/root transforms so it
+    // remains centered on the splat but doesn't move in world space.
+    recalcBound() {
+        const localBound = this.localBound;
+
+        // update splat data local bound
+        const opacity = this.splatData.getProp('opacity');
+        this.splatData.calcAabb(localBound, (i: number) => {
+            return opacity[i] > -1000;
+        });
+
+        // calculate meshinstance aabb (transformed local bound)
+        const meshInstance = this.root.gsplat.instance.meshInstance;
+        meshInstance._aabb.setFromTransformedAabb(localBound, this.entity.getWorldTransform());
```

Comment:
No way to avoid non-API calls?

Reply:
I wasn't able to use public API. This needs addressing with @mvaligursky.

Reply:
Is there an engine issue. Just wanna make sure we don't forget this.

Reply:
Good point, I've created https://github.com/playcanvas/engine/issues/6055.